File: backend/src/main/resources/scraping/session.py
import json

class Session():

    def __init__(self, time, day, theater_name, movie_name, categories):
        self.time = time
        self.day = day
        self.theater = theater_name
        self.movie = movie_name
        self.categories = categories

    def __str__():
        return (f"{self.movie}: {self.day}-{self.time} @{self.theater}")


    def to_json(self):
        return json.dumps(
            self.to_dict(),
            ensure_ascii = False
        )

    def to_dict(self):
        return {
                'cinema': self.theater,
                'movie': self.movie,
                'date': self.day,
                'time': self.time,
                'categories': self.categories
        }

File: backend/src/main/kotlin/com/rotadacultura/model/Session.kt
package com.rotadacultura.model

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.*
import java.io.File

@Serializable
data class Session(
    val cinema: String,
    val movie: String,
    val date: String,
    val time: String,
    val categories: List<String>
) {
    companion object {
        // List to store all sessions' data
        private val sessionsData: List<Session> = Json.decodeFromString(
            File(Cinema::class.java.classLoader.getResource("sessions.json")!!.toURI()).readText()
        )

        fun new(cinema: String, date: String, time: String): Session? {
            return sessionsData.find {
                it.cinema == cinema &&
                it.date == date &&
                it.time == time
            }
        }

        fun getSessionsData(): List<Session> {
            return sessionsData
        }
    }
}
